### ISSUE TITLE
Investigate language decoders

### ISSUE THREAD

**Issue by joshuanianji** [joshuanianji/joshuanianji.github.io#105]
*Note: this is on the `better-language-decoders` branch*

After elm-pages hydrates the static site, some languages change (e.g. Typescript -> Charity). 

I realize the generated `elm.js.opt` contains some data that maps each union type to an integer. The languages that change have integers >= 255, and the resulting language is just a modulo of the original number.

This is definitely not an Elm bug: the languages work fine here (pure client side): https://ellie-app.com/s7nr2SkYVCza1. It seems like the bug is only present when there is a server-client disconnect.

**Comment by joshuanianji** [comment]
UPDATE: Lamdera codec uses int8 to encode the union types. The following was in the compiled elm code:

```
        case 'Lang_tex':
          return $lamdera$codecs$Lamdera$Wire3$encodeSequenceWithoutLength(
            _List_fromArray([
            $lamdera$codecs$Lamdera$Wire3$encodeUnsignedInt8(570)
            ])
          );
```

We need to either set this to 16bits or larger. Might need to contact the lamdera team

**Comment by joshuanianji** [comment]
This is fixed with Lamdera 1.3, just need to wait for it to be released. Not sure if I want to use the -next releases yet
